### PR TITLE
[12.0] Add module base_m2m_custom_field

### PR DESCRIPTION
**Pull request by grindtildeath**
This module adds a new Many2many custom field with a `create_table` attribute.

Many2manyCustom field is useful when a direct access to the relational table
is needed, for example to be editable in a dedicated tree view.

Let's consider following models:

.. code-block:: python

    class MyModelA(models.Model):

        _name = 'my.model.a'

        my_model_b_ids = fields.Many2manyCustom(
            'my.model.b',
            'my_model_a_b_rel',
            'my_model_a_id',
            'my_model_b_id',
            create_table=False,
        )

    class MyModelB(models.Model):

        _name = 'my.model.b'

        my_model_a_ids = fields.Many2manyCustom(
            'my.model.a',
            'my_model_a_b_rel',
            'my_model_b_id',
            'my_model_a_id',
            create_table=False,
        )

    class MyModelABRel(models.Model):

        _name = 'my.model.a.b.rel'

        my_model_a_id = fields.Many2one(
            'my.model.a',
            required=True,
            index=True,
        )
        my_model_b_id = fields.Many2one(
            'my.model.b',
            required=True,
            index=True,
        )

By setting `create_table=False` on the Many2manyCustom field, and using the
relational table name, as `_name` for the relational model, we're able to
define a dedicated tree view for `my.model.a.b.rel`.

.. code-block:: xml

    <record id="my_model_a_b_rel_tree_view" model="ir.ui.view">
        <field name="name">my.model.a.b.rel.tree.view</field>
        <field name="model">my.model.a.b.rel</field>
        <field name="arch" type="xml">
            <tree editable="top">
                <field name="my_model_a_id" />
                <field name="my_model_b_id" />
            </tree>
        </field>
    </record>

### CODE REVIEW

File: base_m2m_custom_field/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+    class MyModelABRel(models.Model):
+
+        _name = 'my.model.a.b.rel'
+
+        my_model_a_id = fields.Many2one(
+            'my.model.a',
+            required=True,
+            index=True,
```
Comment (simahawk) [inline base_m2m_custom_field/readme/USAGE.rst]: add a comment: indexes are mandatory

File: base_m2m_custom_field/fields.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import fields
+
+
+class Many2manyCustom(fields.Many2many):
```
Comment (simahawk) [inline base_m2m_custom_field/fields.py:6]: what about `Many2manyNoTableCreate` (and change module name accordingly)?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: base_m2m_custom_field/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from . import fields
```
Comment (Tonow-c2c) [inline base_m2m_custom_field/__init__.py]:
```suggestion
from . import fields

```

**Review by Tonow-c2c — APPROVED** [review]

File: base_m2m_custom_field/fields.py (hunk 2 above)
Comment (grindtildeath) [reply base_m2m_custom_field/fields.py:6]: IMO this module could be improved with other slots than only `create_table`. That's also why I did set it to True as default...

**Review by simahawk — APPROVED** [review]

**Pull request merged**